torax/_src/geometry/imas.py

```diff
@@ -72,9 +77,27 @@ def geometry_from_IMAS(
     raise ValueError(
         "equilibrium_object must be a string (file path) or an IDS"
     )
-  # TODO(b/431977390): Currently only the first time slice is used, extend to
+  # TODO(b/431977390): Currently only a single time slice is used, extend to
   # support multiple time slices.
-  IMAS_data = equilibrium.time_slice[0]
+  # Convert time to index
+  if slice_time is not None:
+    # Find the closest time in the IDS that is <= slice_time
+    slice_index = (
+        np.searchsorted(equilibrium.time, slice_time, side="right") - 1
+    )
+    if not np.allclose(equilibrium.time[slice_index], slice_time, atol=1e-3):
```

Comment:
I would expect the atol to be very tight here to get an exact match, why was 1e-3 chosen?